src/main/java/org/springframework/data/jdbc/core/EntityRowMapper.java

```diff
@@ -73,8 +74,14 @@ public T mapRow(ResultSet resultSet, int rowNumber) {
 
 		Object id = idProperty == null ? null : readFrom(resultSet, idProperty, "");
 
+		PreferredConstructor<T, RelationalPersistentProperty> persistenceConstructor = entity.getPersistenceConstructor();
```

Comment:
We should avoid the iteration over the properties entirely by returning the just created instance if `PersistentEntity.requiresPropertyPopulation()` is `false`.

Reply:
Done.